### ISSUE TITLE
Instantiating a graph with an empty name will generate invalid dot code

### ISSUE THREAD

**Issue by NeuralNetwork** [erocarrera/pydot#115]
Minimal example:
```python
import pydot

print("with name")
pydot_graph = pydot.Dot("G", graph_type='digraph')
pydot_graph.create()

print("with empty string")
pydot_graph = pydot.Dot("", graph_type='digraph')
pydot_graph.create()
```
Only the first `create()` invocation will succeed. The second will terminate with `Warning: /tmp/tmpw_Wp34: syntax error in line 1 near '{'`

In my opinion the second `create()` invocation should work too.

https://github.com/erocarrera/pydot/blob/master/pydot.py#L1598 is probably the root cause.

**Comment by johnyf** [comment]
Thanks for spotting this, and noting the relevant code. The `show_keyword` attribute that [controls this behavior](https://github.com/erocarrera/pydot/blob/ebfd155d60a6000fedb2169c0d963ed5acfcf0e1/pydot.py#L1598) appears to be a `pydot`-only attribute [set by `dot_parser`](https://github.com/erocarrera/pydot/blob/ebfd155d60a6000fedb2169c0d963ed5acfcf0e1/dot_parser.py#L245).

The below is with graphviz v2.38.0 (20140413.2041) on Debian.

In `dot_parser`, `show_keyword` remembers whether a parsed subgraph started with the keyword `subgraph`. For a subgraph, removing the keyword from input to `dot` also removes the subgraphs themselves (or even creates nodes named after the subgraphs names, say "cluster0" and "cluster1"). So, it seems that it is necessary to dump the `subgraph` keyword. Nonetheless, the "subgraph" keyword is [optional in the dot grammar](96f31c760e0ce99a252107759c67baf6628ae53c), so hiding it will still be supported.

However, in the flattener within `pydot`, the `show_keyword` attribute affects *all* graphs. Giving `{...}` instead of `digraph {...}` as input to `dot` indeed raises a syntax error, as reported.

96f31c760e0ce99a252107759c67baf6628ae53c fixes this, so that only subgraphs be affected.
In addition, the API changes, so that the default now is to show the "subgraph" keyword, if not explicitly specified otherwise.

**Comment by peternowee** [comment]
Was merged as pydot/pydot@cb1435fe, part of release pydot 1.1.0.
